pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -650,3 +747,43 @@ func osdOnSDNFlag(hostnetwork bool, v cephver.CephVersion) []string {
 
 	return args
 }
+
+func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, pvcIndex, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
+	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, pvcIndex, setIndex)
+
+	// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
+	pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
+
+	// Add pvcTemplate name to pvc name. i.e <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>-<pvcTemplateName>
+	if len(pvcTemplate.GetName()) != 0 {
+		pvcGenerateName = pvcGenerateName + pvcTemplate.GetName() + "-"
+	}
+
+	// Add user provided labels to pvcTemplates
+	for k, v := range pvcTemplate.GetLabels() {
+		pvcLabels[k] = v
+	}
+
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: pvcGenerateName,
+			Labels:       pvcLabels,
+			Annotations:  pvcTemplate.Annotations,
+		},
+		Spec: pvcTemplate.Spec,
+	}
+}
+
+func makeStorageClassDeviceSetPVCID(storageClassDeviceSetName string, setIndex, pvcIndex int) (pvcId, pvcLabelSelector string) {
+	pvcStorageClassDeviceSetPVCId := fmt.Sprintf("%s-%v-%v", storageClassDeviceSetName, setIndex, pvcIndex)
```

Comment:
For the pvc ID, let's remove the pvcIndex because there will only ever be one PVC created per template. 
```
fmt.Sprintf("%s-%d", storageClassDeviceSetName, setIndex)
```
So the final generated pvc name will look like:
```
setName-0-templatName-rjf7m
setName-1-templatName-xnl6t
setName-2-templatName-jdb6d
```

Reply:
done